charts/grafana/templates/configmap.yaml

```
@@ -56,6 +56,13 @@ data:
   {{- end -}}
 {{- end -}}
 
+{{- if .Values.alerting }}
+  {{- range $key, $value := .Values.alerting }}
+  {{ $key }}: |
+{{ toYaml $value | indent 4 }}
```

I don't think this would work if the alerting sub yaml files contains self-reference values, e.g. host, port, etc. 

Maybe try this:

```suggestion
{{- tpl $value  . | fromYaml | nindent 4 }}
```

---

I will have a look into that

---

@zanhsieh I had a look and tried it locally. As using `{{ ... }}` is quite common inside the deployed files (eg as `legendFormat: "{{ pod }}"` or as contact points template) a lot of 'escaping' and rewrite in form of ``legendFormat: "{{` {{ pod }} `}}"`` would be necessary.

I think this will increase config and maintenance work for the user of the alerting section, by adding this feature.
How should we proceed here?

---

@mmusenbr 
Did you try / consider tearing down alerts yamls to separate files? See:
https://helm.sh/docs/chart_template_guide/accessing_files/

---

@mmusenbr 
I made some update to my previous suggestion as well as values.yaml and hope to shed some light. Please take a look.

---

> 

Maybe raw string constant?

https://stackoverflow.com/questions/51326021/escape-curly-brackets-in-templates

---

> @zanhsieh I had a look and tried it locally. As using `{{ ... }}` is quite common inside the deployed files (eg as `legendFormat: "{{ pod }}"` or as contact points template) a lot of 'escaping' and rewrite in form of `` legendFormat: "{{` {{ pod }} `}}" `` would be necessary.
> 
> I think this will increase config and maintenance work for the user of the alerting section, by adding this feature. How should we proceed here?

I think we could do raw string.
https://stackoverflow.com/a/51326166/324275

---

@zanhsieh no worries, escaping does not make me uncomfortable, I was just thinking/assuming that a common use-case is to export existing config and provisioning it. And the whole escaping needs to be done and wanted to bring that up. Both ways are fine for me personally.

I will check your latest suggestion and then push it again.